mod-source-record-manager-server/src/test/java/org/folio/services/RawRecordsFlowControlServiceImplTest.java

```diff
@@ -0,0 +1,178 @@
+package org.folio.services;
+
+import org.folio.kafka.KafkaConsumerWrapper;
+import org.folio.services.flowcontrol.RawRecordsFlowControlServiceImpl;
+import org.folio.verticle.consumers.consumerstorage.KafkaConsumersStorage;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.Collections;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.folio.rest.jaxrs.model.DataImportEventTypes.DI_RAW_RECORDS_CHUNK_READ;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RawRecordsFlowControlServiceImplTest {
+
+  @Mock
+  private KafkaConsumersStorage kafkaConsumersStorage;
+
+  @InjectMocks
+  private RawRecordsFlowControlServiceImpl service;
+
+  @Before
+  public void setUp() {
+    ReflectionTestUtils.setField(service, "enableFlowControl", true);
+  }
+
+  @Test
+  public void shouldSkipWhenFlowControlDisabled() {
+    ReflectionTestUtils.setField(service, "enableFlowControl", false);
+    ReflectionTestUtils.setField(service, "maxSimultaneousRecords", 10);
+    ReflectionTestUtils.setField(service, "recordsThreshold", 5);
+
+    service.trackChunkReceivedEvent(10);
+    service.trackChunkDuplicateEvent(10);
+    service.trackRecordCompleteEvent();
+
+    verifyNoMoreInteractions(kafkaConsumersStorage);
```

Comment:
Just for my understanding...

Please explain why any interaction would occur with the mock at this point. I think it is the `verifyNoMoreInteractions` that is confusing me. if it was `verifyNoInteractions` it would make more sense to me haha, but you dont control that.

 So with all the tracking done, no interaction should occur with `kafkaConsumerStorage` because flow control is disabled?

Reply:
Thanks, changed to verifyNoInteractions and added more comments in code to clarify this test.
The answer to the second question - yes, no interaction should occur with kafkaConsumerStorage because flow control is disabled